File: triangle.go
package main

import (
    "bufio"
    "fmt"
    "os"
    "strconv"
)

func printTriangle(rows int) {
    for i := 1; i <= rows; i++ {
        // Print spaces before stars
        for j := 1; j <= rows-i; j++ {
            fmt.Print(" ")
        }

        // Print stars
        for k := 1; k <= i; k++ {
            fmt.Print("* ")
        }

        // Print newline
        fmt.Println()
    }
}

func main() {
    // Ask the user for the number of rows
    scanner := bufio.NewScanner(os.Stdin)
    fmt.Print("Enter the number of rows: ")
    scanner.Scan()
    rows, _ := strconv.Atoi(scanner.Text())

    // Print the triangle pattern
    printTriangle(rows)
}

File: triangle.r
print_triangle <- function(rows) {
  for (i in 1:rows) {
    # Print spaces before stars
    cat(rep(" ", rows - i))
    
    # Print stars
    cat(rep("* ", i), "\n")
  }
}

# Ask the user for the number of rows
rows <- as.integer(readLines("stdin", n = 1))

# Print the triangle pattern
print_triangle(rows)